NuRadioMC/EvtGen/generator.py

```diff
@@ -1370,8 +1370,8 @@ def generate_eventlist_cylinder(filename, n_events, Emin, Emax,
                                 n_interaction += 1
                                 data_sets_fiducial['shower_energies'][-1] = product.energy
                                 data_sets_fiducial['inelasticity'][-1] = np.nan
-                                # interaction_type is either 'had' or 'em' for proposal products
-                                data_sets_fiducial['interaction_type'][-1] = product.shower_type
+                                # interaction_type is particle codes taken from NuRadioProposal.py
+                                data_sets_fiducial['interaction_type'][-1] = product.code
```

Comment:
I don't think this line does what we want.
The `interaction_type` being `cc` or `nc` might play a role in what the Askaryan model makes out of it - but I haven't checked.


 Was unaware that you are taking over implementing these changes. @jannavischer in that case please coordinate and make sure you are not doing things twice.
 
  I think you want to get the flavor of the `parent` not the product. Might make sence to wait until the proposal 7 branch is merged - since the interface changes

Reply:
For the interaction type I thought we want the particle code of the produce (to know what initiated the shower). The flavour/code of  the parent is stored in the item "flavor"

@jannavischer, sorry I did not know that you were working on this. As part of writing a tool to convert nur to hdf5 files I thought starting with this actually. I am currently trying to understand how to get the energy of the muon and tau from proposal exactly before the energy loss

> The interaction_type being cc or nc might play a role in what the Askaryan model makes out of it - but I haven't checked.

Are you sure? Or do you confuse it with shower_type (which has to be em or had).

Reply:
> Might make sence to wait until the proposal 7 branch is merged - since the interface changes

hm very good point.